File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        boolean flag = x < 0;
        String str = String.valueOf(Math.abs(x));
        StringBuilder sb = new StringBuilder(str).reverse();
        try{
            res = Integer.parseInt(sb.toString());
        } catch (NumberFormatException e) {
            return 0;
        }
        return flag ? -res : res;

    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        flag = False
        if x < 0:
            x = -x
            flag = True
        while(x>0):
            rem = x%10
            result = result * 10 + rem
            x //= 10
        if result > 2 ** 31 - 1:
            return 0
        if flag:
            return -result
        return result